test/multiple-user-principal-types.test.js

```diff
@@ -4,27 +4,41 @@
 // License text available at https://opensource.org/licenses/MIT
 
 'use strict';
-var assert = require('assert');
 var expect = require('./helpers/expect');
 var request = require('supertest');
 var loopback = require('../');
+var ctx = require('../lib/access-context');
+var AccessContext = ctx.AccessContext;
+var Principal = ctx.Principal;
 var Promise = require('bluebird');
-var async = require('async');
 
-describe('Multiple users with custom principalType:', function() {
+describe('Multiple users with custom principalType', function() {
   this.timeout(10000);
 
   var commonCredentials = {email: 'foo@bar.com', password: 'bar'};
-  var OneUser, AnotherUser, AccessToken, userFromOneModel, userFromAnotherModel;
+  var app, OneUser, AnotherUser, AccessToken, Role,
+    userFromOneModel, userFromAnotherModel, userRole, userOneBaseContext;
+
+  // helpers
+  function createUserModel(app, name, options) {
+    var model = app.registry.createModel(Object.assign({name: name}, options));
+    app.model(model, {dataSource: 'db'});
+    model.setMaxListeners(0); // allow many User.afterRemote's to be called
+    return model;
+  }
+
+  function waitForEvent(emitter, name) {
+    return new Promise(function(resolve, reject) {
+      emitter.once(name, resolve);
+    });
+  };
 
-  // Create a local app variable to prevent clashes with the global
-  // variable shared by all tests. While this should not be necessary if
-  // the tests were written correctly, it turns out that's not the case :(
-  var app = null;
+  function getIds(array) {
+    return array.map(function(it) { return it.id; });
+  };
```

Comment:
Our convention is to put helpers at the bottom, see http://loopback.io/doc/en/contrib/style-guide.html